Source/WebCore/editing/TextIterator.cpp

```diff
@@ -466,7 +478,13 @@ static bool isConsideredSkippedContent(const RenderBox* renderBox, TextIteratorB
     if (!renderBox || !isSkippedContentRoot(*renderBox))
         return false;
 
-    return behaviors.contains(TextIteratorBehavior::EntersSkippedContentRelevantToUser) ? renderBox->style().usedContentVisibility() == ContentVisibility::Hidden : true;
+    if (behaviors.contains(TextIteratorBehavior::EntersSkippedContentRelevantToUser)) {
+        if (renderBox->document().settings().detailsAutoExpandEnabled() && renderBox->style().autoRevealsWhenFound())
+            return false;
+        return renderBox->style().usedContentVisibility() == ContentVisibility::Hidden;
```

Comment:
We should do the content visibility check first, since it’s usually not hidden.

Why does `autoRevealsWhenFound` need to be precomputed and stored on the RenderStyle? Why not just check         `m_document->settings().detailsAutoExpandEnabled() && is<HTMLDetailsElement>(element)` here, through an appropriate helper function? Similarly, why do the `detailsAutoExpandEnabled()` check twice?

Reply:
Same here for the bit. Though I can move the content-visibility check first.